Resources/Prototypes/Nyanotrasen/Entities/Mobs/NPCs/fungi.yml

```diff
@@ -0,0 +1,82 @@
+- type: entity
+  parent: SimpleSpaceMobBase
+  id: MobFungusBase
+  name: fungus
+  abstract: true
+  components:
+  - type: Bloodstream
+    bloodReagent: Mold
+  - type: PsionicInsulation
+
+- type: entity
+  parent: MobFungusBase
+  id: MobFungusGasSpore
+  name: gas spore
+  description: A floating ball of gas.
+  components:
+  - type: CombatMode
+    combatToggleAction:
+      enabled: false
+      autoPopulate: false
+      name: action-name-combat
+  - type: GasCanister
+    gasMixture:
+      volume: 200
+      moles:
+        - 0 # oxygen
+        - 0 # nitrogen
+        - 0 # CO2
+        - 0 # Plasma
+        - 0 # Tritium
+        - 0 #  Water vapor
+        - 300 # Miasma
+      temperature: 293.15
```

Comment:
This component is for a mechanical gas canister and has assumptions about it related to that, i.e. you could theoretically hook a gas tank up to this entity, or interact with a UI, if you can click on it.

Is this just for the DumpCanisterBehavior? Maybe this information could be put on the SelfExploder and a separate behavior created.

Reply:
the component does too much. I think I can write this and then make an issue to (1) remove it, (2) decouple GasCanister from that behavior, (3) make a "GasContainer" component that just holds the gas mixture, (4) remove it from the portable scrubbers and (5) upstream that change.